### ISSUE TITLE
Tomb contents commands in README.md don't work with binary keys

### ISSUE THREAD

**Issue by grcancelliere** [dyne/Tomb#372]
while `zsh` handles binary data correctly in its process substitution, other shells like `bash` and `fish` don't.

`zsh`:
```
% print -R -n - $pass | wc -c
509
```

`bash`:
```
-bash: warning: command substitution: ignored null byte in input
$ echo -n -e $pass | wc -c
507
```

`fish`:
```
> echo -n -e $pass | wc -c
323
```

In order to pass the correct key to `cryptsetup` I think the best way would be to pipe the `gpg` output to `head -c -1` to remove the newline created by `gpg`; this way the commands would be:

```
lo=$(losetup -f)
losetup -f secret.tomb
gpg -d secret.key | head -c -1 | cryptsetup --key-file - luksOpen $lo secret
mount /dev/mapper/secret /mnt
```
**edit 1**

That test was with a gpg key; today i tried with symmetric encryption and the resulting file from gpg didn't have a newline, so the commands above wouldn't work... maybe this is a gpg bug

**edit 2**

It's definitely not a gpg bug, in `gen_key()` the `print` options are just `-` for a gpg key ([line 1423](https://github.com/dyne/Tomb/blob/master/tomb#L1423) and `-n -` for a symmetric key [line 1498](https://github.com/dyne/Tomb/blob/master/tomb#L1498)

**Comment by jaromil** [comment]
Thanks!

It may be enough of a fix to use `-n` everywhere, yet this should be tested.

I'm waiting for #388

**Comment by j-m-t** [comment]
I wanted to chime in and also say that the code snippet in the README would not work for me:

```
echo -n -e "$pass" | sudo cryptsetup --key-file - luksOpen $lo secret
```
would produce the following:
```
No key available with this passphrase.
```
The inclusion of the `-n` argument with `echo` didn't format the decrypted key properly, at least not when using `bash`. For what its worth, I'm using `bash` version 5.0.3.

The modification by @grcancelliere  did work for me though:
```
gpg -d .password.tomb.key | head -c -1 | cryptsetup --key-file - luksOpen $lo secret
```

Also, I think the closed issue #289 may be related to this.

Besides this issue, the instructions in the README were very helpful in opening up a corrupted tomb and fixing it with `fsck`.  Many thanks @jaromil  for the work you and others have put into this.
